app/derivative_services/audio_derivative_service.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+class AudioDerivativeService
+  class Factory
+    attr_reader :change_set_persister
+    delegate :metadata_adapter, :storage_adapter, to: :change_set_persister
+    delegate :query_service, to: :metadata_adapter
+    def initialize(change_set_persister:)
+      @change_set_persister = change_set_persister
+    end
+
+    def new(change_set)
+      AudioDerivativeService.new(change_set: change_set, change_set_persister: change_set_persister, target_file: target_file(change_set.resource))
+    end
+
+    def target_file(resource)
+      resource.original_file
+    end
+  end
+
+  attr_reader :change_set, :change_set_persister, :target_file
+  delegate :mime_type, to: :target_file
+  delegate :resource, to: :change_set
+  delegate :storage_adapter, to: :change_set_persister
+  def initialize(change_set:, change_set_persister:, target_file:)
+    @change_set = change_set
+    @change_set_persister = change_set_persister
+    @target_file = target_file
+  end
+
+  def valid?
+    ["audio/x-wav"].include?(mime_type.first)
+  end
+
+  def create_derivatives
+    progress = ActiveEncode::Base.create(file_object.disk_path, outputs: ffmpeg_outputs)
+    sleep(1 / 1000 * 50) until progress.reload.completed? || progress.failed?
```

Comment:
Is there any reason to poll at a sub-millisecond frequency?

Reply:
Oh, I was assuming the `*` had precedence over `/` — could this be `50 / 1000` for readability?